src/test/scala/firrtlTests/annotationTests/UnrecognizedAnnotationSpec.scala

```diff
@@ -0,0 +1,344 @@
+    val fileNames = setupFiles(addAllowUnrecognizedFlag = true, addAllowUnrecognizedAnno = true)
+    shouldSucceed(fileNames)
+  }
+
+  def shouldSucceed(fileNames: TestFileNames): Unit = {
+    val args = makeCommandLineArgs(fileNames)
+    FirrtlMain.main(args)
+
+    val outputAnnotationText = FileUtils.getText(fileNames.outputAnnotationsFull)
+    outputAnnotationText should include("freechips.rocketchip.util.RegFieldDescMappingAnnotation")
+    outputAnnotationText should include("freechips.rocketchip.util.SRAMAnnotation")
+  }
+
+  case class TestFileNames(
+    allowUnrecognized:     Boolean,
+    inputAnnotations:      String,
+    outputAnnotations:     String,
+    outputAnnotationsFull: String,
+    firrtlSource:          String,
+    firrtlOutput:          String)
+
+  def setupFiles(addAllowUnrecognizedFlag: Boolean, addAllowUnrecognizedAnno: Boolean): TestFileNames = {
+    val dirName = (addAllowUnrecognizedFlag, addAllowUnrecognizedAnno) match {
+      case (false, false) => s"test_run_dir/unrecognized_annotation_fail"
+      case (true, false)  => s"test_run_dir/unrecognized_annotation_flag"
+      case (false, true)  => s"test_run_dir/unrecognized_annotation_anno"
+      case (true, true)   => s"test_run_dir/unrecognized_annotation_flag_and_anno"
+    }
+    val dir = new File(dirName)
+    dir.mkdirs()
+
+    val fileNames = TestFileNames(
+      allowUnrecognized = addAllowUnrecognizedFlag,
+      inputAnnotations = s"$dirName/input_annotations.json",
+      outputAnnotations = s"$dirName/output_annotations",
+      outputAnnotationsFull = s"$dirName/output_annotations.anno.json",
+      firrtlSource = s"$dirName/trivial.fir",
+      firrtlOutput = s"$dirName/trivial_out"
+    )
+
+    def writeText(fileName: String, text: String): Unit = {
+      val writer = new PrintWriter(fileName)
+      writer.write(text)
+      writer.close()
+    }
+
+    writeText(
+      fileNames.inputAnnotations,
+      UnrecognizedAnnotationTextGenerator.jsonText(includeAllowUnrecognizedAnnotations = addAllowUnrecognizedAnno)
+    )
+    writeText(
+      fileNames.firrtlSource,
+      s"""
+         |circuit Trivial :
+         |  module Trivial :
+         |    input clock : Clock
+         |    input reset : UInt<1>
+         |""".stripMargin
+    )
+    fileNames
+  }
+
+  /* construct an array of command line strings, based on file names and  */
+  def makeCommandLineArgs(fileNames: TestFileNames): Array[String] = {
+
+    (if (fileNames.allowUnrecognized) {
+       Array("--allow-unrecognized-annotations")
+     } else {
+       Array.empty[String]
+     }) ++
+      Array(
+        "--annotation-file",
+        fileNames.inputAnnotations,
+        "-i",
+        fileNames.firrtlSource,
+        "-X",
+        "high",
+        "-o",
+        fileNames.firrtlOutput,
+        "--output-annotation-file",
+        fileNames.outputAnnotations
+      )
+  }
+}
+
+object UnrecognizedAnnotationTextGenerator {
```

Comment:
I get using real test cases but this is like 170 lines of stuff that doesn't really add anything. I think it's be better just to make up 1 or two small, artificial annotations and leave it at that.

Reply:
Sure. Reduced the cruft to just the class names